### ISSUE TITLE
Dialog for "SQL Cipher encryption" on read access failure

### ISSUE THREAD

**Issue by AndreasMartin72** [sqlitebrowser/sqlitebrowser#866]
### Details for the issue

When trying to open a database that is locked by another application, I get the "SQL Cipher encryption" dialog which hides the real issue.

#### I'm opening this issue because:

- [ ] DB4S is crashing
- [ ] DB4S has a bug
- [ ] DB4S needs a feature
- [x] DB4S has another problem

#### I'm using DB4S on:

- [x] Windows: ( _version:_ 7 )
- [ ] Linux: ( _distro:_ ___ )
- [ ] Mac OS: ( _version:_ ___ )
- [ ] Other: ___

#### I'm using DB4S version:

- [x] 3.9.1
- [ ] 3.9.0
- [ ] Other: ___

#### I have also:

- [ ] Tried out the nightly version: https://github.com/sqlitebrowser/sqlitebrowser#windows
- [x] Searched for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
Ahh interesting.  Yep, it makes sense we should handle this better. :smile:

**Comment by MKleusberg** [comment]
Can you try tomorrow's nightly build, @AndreasMartin72? I've just changed the error handling code a bit but I'm not entirely sure it affects your problem. Still, it's worth a try :smiley:

**Comment by abbasshah17** [comment]
Hi, I think I might be facing a similar issue, I pulled SQLCipher DB (created by an Android App) from an Android Box and tried to open, getting the same "Please enter the key used to encrypt the database" dialog over and over again, even though I have supplied it with the correct passkey. I am not setting page size, so I put page size 1024 in the dialog.

Could this be the same issue or a different one? If this is then please tell me how to _unlock_  the database file.

**Comment by MKleusberg** [comment]
@abbasshah17 Hard to say what's the problem there. We have no chance of finding out what exactly is wrong when failing to open an encrypted database. What sort of Android app are you using? Maybe we can find out what encryption settings they are using.

**Comment by OmgImAlexis** [comment]
I get this issue too when trying to open a file that's on a samba share yet when I copy the file to my desktop and open it there's no issues.

**Comment by justinclift** [comment]
@OmgImAlexis Interesting.  Does it still happen with our recent nightly builds?

**Comment by OmgImAlexis** [comment]
I'm not sure as I have never used this product before and only noticed it on the first time I tried it.

Do you have a link to the nightly builds that I can try?
I'm using Mac OS Sierra if that matters.

**Comment by justinclift** [comment]
Oops, I should have included it. :smile:

These are our nightly builds for OSX:

&nbsp; &nbsp; https://nightlies.sqlitebrowser.org/osx/

They're ordered by date.  There's a build that includes encryption, and one without.  If your database isn't encrypted, then either will be suitable for trying out.

**Comment by OmgImAlexis** [comment]
@justinclift both are working fine for me now. So the nightly has whatever issue I had fixed.

**Comment by justinclift** [comment]
Excellent @OmgImAlexis. :smile:

**Comment by justinclift** [comment]
@AndreasMartin72 Ping, did you have time to try out a recent nightly build? 😄

**Comment by justinclift** [comment]
@abbasshah17 Did you ever get this fixed?

**Comment by justinclift** [comment]
Closing this due to lack of response.

**Comment by abbasshah17** [comment]
Apologies for the late response, I received the mail on a vacation when I had no tested news to add. But @justinclift unfortunately yes! I got it fixed but I don't know how. I didn't update the software, so unless some windows `exe`, `dll`, etc. files were required to run which were missing from the system previously I have no idea why DB Browser is suddenly working. Anyway the other day a junior colleague opened it because he didn't know DB Browser wasn't working for SQLCipher.

Meanwhile I checked and rechecked versions on both machines, same version (Software and OS both). Opened the file myself and voila, now it was opening the file as innocently as possible.
